OrangeDesignSystemDemo/OrangeDesignSystemDemoTests/AppNewsListViewModelTests.swift

```diff
@@ -0,0 +1,176 @@
+//
+// Software Name: Orange Design System (iOS)
+// SPDX-FileCopyrightText: Copyright (c) 2021 - 2023 Orange SA
+// SPDX-License-Identifier: MIT
+//
+// This software is distributed under the MIT license.
+//
+
+import Foundation
+@testable import OrangeDesignSystem
+import XCTest
+
+final class AppNewsListViewModelTests: XCTestCase {
+
+    // ==================
+    // MARK: - Test cases
+    // ==================
+
+    func testAppNewsListViewModel_newInstance() {
+        // Given
+        let viewModel = AppNewsListViewModel(fromFile: "")
+
+        // When
+        let state = viewModel.releaseDescriptions
+
+        // Then
+        var testPassed = false
+        switch state {
+        case .loading:
+            testPassed = true
+        default:
+            testPassed = false
+        }
+        XCTAssertTrue(testPassed)
+    }
+
+    func testAppNewsListViewModel_load_noExistingFile() {
+        // Given
+        let viewModel = AppNewsListViewModel(fromFile: "polynectar.recette")
+
+        // When
+        viewModel.load()
+        let state = viewModel.releaseDescriptions
+
+        // Then
+        switch state {
+        case let .error(error):
+            XCTAssertTrue(error == AppNewsListViewModel.Error.resourceNotFound, "Supposed to have an AppNewsListViewModel.Error.resourceNotFound instead of \(error)")
+        default:
+            XCTFail("Supposed to have an AppNewsListViewModel.Error.resourceNotFound")
+        }
+    }
+
+    func testAppNewsListViewModel_load_noUTF8ContentInFile() {
```

Comment:
why without utf8?
if it is a jpeg is not a json, so it failed (with or without utf8)

Reply:
In fact we parse the JSON file by reading the content as a string content, in UTF8.
If this step failed, an error is managed.
This unit test is here to deal that particular case, so I tested with a small JPEG file which will make the parsing fail (because it cannot be parsed as a String in UTF8à.